pkg/kinds/dashboard/validator.go

```diff
@@ -0,0 +1,82 @@
+package dashboard
+
+import (
+	"context"
+	_ "embed"
+	json "encoding/json"
+	fmt "fmt"
+	"strings"
+	"sync"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/cuecontext"
+	"cuelang.org/go/cue/errors"
+	cuejson "cuelang.org/go/encoding/json"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+)
+
+// ValidateDashboardSpec validates a JSON object against the dashboard kind schema.
+func ValidateDashboardSpec(ctx context.Context, obj common.Unstructured) field.ErrorList {
```

Comment:
@ryantxu @marcoabreu I think this could technically live in `apps/dashboard/pkg/apis/v1alpha1` and we could have a similar function in `v2alpha1` which'd use CUE from v2, because current codegen loses some of the cool validation CUE can do when going from CUE to Go to OpenAPI.

Just thinking out loud.